stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/PopUtil.java

```diff
@@ -0,0 +1,213 @@
+    private static final int MAX_PORT = 65535;
+
+    /**
+     * Generates Properties object using the passed MapValue.
+     *
+     * @param popConfig MapValue with the configuration values
+     * @param isPop True if the protocol is POP3 and false otherwise (if protocol is IMAP)
+     * @return Properties object
+     */
+    public static Properties getProperties(MapValue popConfig, boolean isPop) {
+        if (log.isDebugEnabled()) {
+            log.debug("[PopUtil][getProperties] Extracting Property values from Config");
+        }
+        Properties properties = new Properties();
+        if (isPop) {
+            properties.put(PopConstants.PROPS_POP_HOST, popConfig.getStringValue(PopConstants.PROPS_HOST));
+            properties.put(PopConstants.PROPS_POP_PORT,
+                    extractPortValue(popConfig.getIntValue(PopConstants.PROPS_PORT), isPop));
+            properties.put(PopConstants.PROPS_POP_AUTH, "true");
+            properties.put(PopConstants.PROPS_POP_STARTTLS, "true");
+            properties.put(PopConstants.PROPS_POP_SSL_ENABLE, popConfig.getBooleanValue(PopConstants.PROPS_SSL));
+            properties.put(EmailConstants.MAIL_STORE_PROTOCOL, PopConstants.POP_PROTOCOL);
+        } else {
+            properties.put(ImapConstants.PROPS_IMAP_HOST, popConfig.getStringValue(PopConstants.PROPS_HOST));
+            properties.put(ImapConstants.PROPS_IMAP_PORT,
+                    extractPortValue(popConfig.getIntValue(PopConstants.PROPS_PORT), isPop));
+            properties.put(ImapConstants.PROPS_IMAP_AUTH, "true");
+            properties.put(ImapConstants.PROPS_IMAP_STARTTLS, "true");
+            properties.put(ImapConstants.PROPS_IMAP_SSL_ENABLE, popConfig.getBooleanValue(PopConstants.PROPS_SSL));
+            properties.put(EmailConstants.MAIL_STORE_PROTOCOL, ImapConstants.IMAP_PROTOCOL);
+        }
+
+        properties.put(PopConstants.PROPS_USERNAME, popConfig.getStringValue(PopConstants.PROPS_USERNAME));
+        properties.put(PopConstants.PROPS_PASSWORD, popConfig.getStringValue(PopConstants.PROPS_PASSWORD));
+
+        if (log.isDebugEnabled()) {
+            log.debug("[PopUtil][getProperties] Property object created. Returning Object to PopClient");
+        }
+        return properties;
+    }
+
+    /**
+     * Generates MapValue object using the passed message.
+     *
+     * @param message received
+     * @return MapValue object
+     * @throws MessagingException If an error occurs related to messaging
+     * @throws IOException If an error occurs related to I/O
+     */
+    public static MapValue getMapValue(Message message) throws MessagingException, IOException {
+        if (log.isDebugEnabled()) {
+            log.debug("[PopUtil][getMapValue] Extracting MapValue values from Config");
+        }
+        Map<String, Object> valueMap = new HashMap<>();
+        Address[] toAddress = message.getRecipients(Message.RecipientType.TO);
```

Comment:
We can break this into smaller methods that return Address[]? We can avoid this longer method IMO.

Reply:
Broken into methods.